src/ImageSharp.Web/Providers/PhysicalFileSystemProvider.cs

```diff
@@ -101,5 +58,18 @@ internal static string GetProviderRoot(PhysicalFileSystemProviderOptions provide
 
             return PathUtilities.EnsureTrailingSlash(providerRootPath);
         }
+
+        private static PhysicalFileProvider GetProvider(
+            PhysicalFileSystemProviderOptions options,
+#if NETCOREAPP2_1
+            IHostingEnvironment environment)
+#else
+            IWebHostEnvironment environment)
+#endif
+        {
+            Guard.NotNull(options, nameof(options));
+            Guard.NotNull(environment, nameof(environment));
+            return new(GetProviderRoot(options, environment.WebRootPath, environment.ContentRootPath));
```

Comment:
Keep in mind that the `PhysicalFileProvider` requires the directory to exist (it throws an exception if not). I guess that's fine, because you don't want to create directories in constructors (would be a nightmare when used in tests) and this provider is otherwise useless anyway...

If you have a custom provider root set (to a fully qualified path), you can also use the configure callback to ensure the directory exists before the constructor is called:

```c#
.Configure<PhysicalFileSystemProviderOptions>(options =>
{
    if (!string.IsNullOrEmpty(options.ProviderRootPath))
    {
        Directory.CreateDirectory(options.ProviderRootPath);
    }
})
```